server/plugin.go

```diff
@@ -357,6 +337,43 @@ func (p *Plugin) start(isRestart bool) {
 	if err = p.API.RegisterCommand(p.createCommand(p.getConfiguration().SyncLinkedChannels)); err != nil {
 		p.API.LogError("Failed to register command", "error", err)
 	}
+
+	gmsDmsAutoconnectMutex, err := cluster.NewMutex(p.API, gmsDmsAutoconnectClusterMutexKey)
+	if err != nil {
+		p.API.LogError("Unable to create mutex for automatic connection of dms and gms", "error", err.Error())
+		return
+	}
+
+	gmsDmsAutoconnectMutex.Lock()
+	defer gmsDmsAutoconnectMutex.Unlock()
+
+	for {
+		var ids []string
+		ids, err = p.store.ListDMsGMsToConnectBatch()
+		if err != nil {
+			p.API.LogWarn("Unable to list the dms/gms to connect", "error", err.Error())
+			continue
+		}
+		if len(ids) == 0 {
+			p.API.LogInfo("DMs and GMs automatic connection finished")
+			break
+		}
+		for _, id := range ids {
+			if _, err = p.API.ShareChannel(&model.SharedChannel{
+				ChannelId: id,
+				Home:      true,
+				CreatorId: p.userID,
+				ShareName: id,
+			}); err != nil {
+				p.API.LogWarn("Unable to share channel", "channel_id", id, "error", err.Error())
+			}
+			if err = p.inviteRemoteToChannel(id, p.remoteID, p.userID); err != nil {
+				p.API.LogWarn("Unable simulate the invite remote channel", "channel_id", id, "error", err.Error())
+			}
+		}
+		// Give some time to other work to happen in the server
+		time.Sleep(1 * time.Second)
```

Comment:
On a server with 10,000 users, there's theoretically up to 100,000,000 direct message channels alone. Even if we assume only a 5% connected graph, that's still `5,000,000` channels. With a batch size of `500`, this will still take 10,000 seconds or or almost 3 hours.

This is fine for our bug bash, but let's not merge this.

Reply:
> can be 3 hours, can be 10 hours, it doesn't matter that much

This is fine for testing, but we absolutely can't ship something with this much variability to production. I'd propose we wait for v9.7 and shared channels doing this natively (and efficiently) instead before considering this for production.

Reply:
As far as I understand, this can be completely unneeded in the future with some shared channels code changes.

Reply:
https://github.com/mattermost/mattermost/pull/26753 is merged.   It will share any DM/GM not previously shared on next post.